python/pyspark/tests/test_taskcontext.py

```diff
@@ -181,6 +227,40 @@ def context_barrier(x):
         for pid in pids:
             self.assertTrue(pid in worker_pids)
 
+    def test_task_context_correct_with_python_worker_reuse(self):
+        """Verify the task context correct when reused python worker"""
+        # start a normal job first to start all workers and get all worker pids
+        worker_pids = self.sc.parallelize(xrange(2), 2).map(lambda x: os.getpid()).collect()
+        # the worker will reuse in this barrier job
+        rdd = self.sc.parallelize(xrange(10), 2)
+
+        def context(iterator):
+            tp = TaskContext.get().partitionId()
+            try:
+                bp = BarrierTaskContext.get().partitionId()
+            except Exception:
+                bp = -1
+
+            yield (tp, bp, os.getpid())
+
+        def verify(result, task_context_target, barrier_context_target):
+            tps = list(map(lambda x: x[0], result))
+            bps = list(map(lambda x: x[1], result))
+            pids = list(map(lambda x: x[2], result))
```

Comment:
`[[tps], [bps], [pids]] = result`

Reply:
result is a list(tuple), maybe  `tps, bps, pids = zip(*result)` if you think current too noise.

Reply:
Yeah, either way should be fine.

Reply:
BTW, iterable unpacking works with list(tuple) too

Reply:
Not very familiar with python, I tried but got failed.
```
In [15]: result = [(1, 2, 3), (4, 5, 6)]                                                                                                                                                

In [16]: [[a], [b], [c]] = result                                                                                                                                                       
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-20ce771eb68a> in <module>
----> 1 [[a], [b], [c]] = result

ValueError: not enough values to unpack (expected 3, got 2)
```